server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -154,6 +160,7 @@ public class RestoreService implements ClusterStateApplier {
 
     // It's OK to change some settings, but we shouldn't allow simply removing them
     private static final Set<String> UNREMOVABLE_SETTINGS;
+    private static final int REMOTE_DATA_TO_FILE_CACHE_SIZE_RATIO = 5;
```

Comment:
I think this needs to be a cluster setting, with the default to "no limit". Otherwise it is a backwards incompatible behavior change.